tb_plugin/torch_tb_profiler/profiler/op_tree.py

```diff
@@ -0,0 +1,329 @@
+import sys
+from collections import defaultdict
+
+from .. import utils
+from .node import (BackwardNode, ModuleNode, OperatorNode, ProfilerStepNode,
+                   RuntimeNode, is_operator_node)
+from .trace import EventTypes
+
+logger = utils.get_logger()
+
+
+class OpTreeBuilder:
+    BACKWARD_ROOT_PREFIX = 'autograd::engine::evaluate_function:'
+    BACKWARD_ACCUMULATE_GRAD = 'autograd::engine::evaluate_function: torch::autograd::AccumulateGrad'
+
+    def __init__(self, tid2list, tid2zero_rt_list, corrid_to_device, fwd_bwd_map):
+        self.tid2list = tid2list
+        self.tid2zero_rt_list = tid2zero_rt_list
+        self.corrid_to_device = corrid_to_device
+        self.fwd_bwd_map = fwd_bwd_map
+        self.main_tid = None
+        self.tid2tree = None
+
+    def build_tree(self):
+        '''Construct the BackwardNode and replace the original backward nodes
+        '''
+        self.tid2tree = self._build_tree(self.tid2list, self.tid2zero_rt_list, self.corrid_to_device)
```

Comment:
self.tid2list, self.tid2zero_rt_list, self.corrid_to_device is all members of this class. They need not to pass through argument.

Reply:
will move as build_tree parameters